src/AspNetCore.Identity.MongoDB/UserClaimsPrincipalFactory.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace AspNetCore.Identity.MongoDB
+{
+    public class UserClaimsPrincipalFactory<TUser> : IUserClaimsPrincipalFactory<TUser>
```

Comment:
Can we rename this to something like `ClaimsBasedUserClaimsPrincipalFactory` so that it wouldn't clash with [the one which is in Identity library itself](https://github.com/aspnet/Identity/blob/1.0.0/src/Microsoft.AspNetCore.Identity/UserClaimsPrincipalFactory.cs). I know that the generic parameters makes it distinguishable but it could still cause some confusion.